app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -223,6 +224,8 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         if (settings().isMarketingTelemetryEnabled) {
             components.analytics.metrics.start(MetricServiceType.Marketing)
         }
+
+        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(settings().isTelemetryEnabled)
```

Comment:
One issue I have is currently we don't have a preference switch for crash reporting because crashes are manually sent when the user tap on the action in the crash notification.  So I used the telemetry preference.